### PR TITLE
chore(pyodbc): set db.system and db.user span tags

### LINKED ISSUE #7104: dd-trace-py + pyodbc (mssql): missing spans
### Summary of problem

I have a python application connected to mssql server using pyodbc.

I realized that in the generated dd traces I can't find some important spans. For example: db.system or db.type, db.user, db.operation.

It seems that when I connect my application to other DB like postgres or mysql the spans are generate well 

### Which version of dd-trace-py are you using?

My environment:
- python 3.9
- flask 2.2.4
- dd-trace-py 1.20.2

### Which libraries and their versions are you using?

<details>
  <summary>`pip freeze`</summary>
  
</details>

### PR DESCRIPTION
**Pull request by emmettbutler**
This change fixes #7104 by setting the `db.system` and `db.user` span tags from the pyodbc integration.

`db.user` is not tested because the existing test harness uses a sqlite database, which does not have the concept of a username.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: tests/contrib/pyodbc/test_pyodbc.py [hunk 1]
```diff
@@ -282,6 +282,8 @@ def test_patch_unpatch(self):
             assert span.service == "pyodbc"
             assert span.name == "pyodbc.query"
             assert span.span_type == "sql"
+            assert span.get_tag("db.system") == "pyodbc"
```
Comment (robertomonteromiguel) [inline tests/contrib/pyodbc/test_pyodbc.py:285]: The value for "db.system" span, must be an identifier for the DBMS. For example, "mssql"?
**DD doc**: https://datadoghq.atlassian.net/wiki/spaces/APM/pages/2357395856/Span+attributes#db.system
**Otel doc**: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]

**Review by robertomonteromiguel — APPROVED** [review]

File: tests/contrib/pyodbc/test_pyodbc.py [hunk 2]
```diff
@@ -282,6 +282,8 @@ def test_patch_unpatch(self):
             assert span.service == "pyodbc"
             assert span.name == "pyodbc.query"
             assert span.span_type == "sql"
+            assert span.get_tag("db.system") == "SQLite"
+            assert span.get_tag("db.user") == ""
```
Comment (mabdinur) [inline tests/contrib/pyodbc/test_pyodbc.py:286]: nit: can we update this test so that it sets a user name

**Comment by robertomonteromiguel** [comment]
when this PR is merged we can activate the tests:

https://github.com/DataDog/system-tests/blob/4456488751422ed89a279aa3dbfaf7043a76522a/tests/integrations/test_db_integrations_sql.py#L522

File: tests/contrib/pyodbc/test_pyodbc.py (hunk 2 above)
Comment (emmettbutler) [reply tests/contrib/pyodbc/test_pyodbc.py:286]: Doing so requires changing the test harness to use a database other than sqlite. Do you think that's worth the effort?

**Pull request merged**